### PR TITLE
 Drop @ratelimit Annotation, Create #[RateLimit] Attribute, Drop php7 support, Drop symfony 3 & 4 support

### LINKED ISSUE #130: sensio/framework-extra-bundle is deprecated
`sensio/framework-extra-bundle` is deprecated, however this bundle uses it to handle the `@RateLimit` annotation.

Are you ok if I send a pull request in which:
- drop the  `@RateLimit` annotation in favor of a PHP attribute?
- drop php 7 support
- drop symfony 3.x and 4.x support?

if this is merged will require a major release

(i'm also offering my help to maintain this bundle)

### PR DESCRIPTION
**Pull request by axi**
Should fix #130.

There is still (at least) one thing I would like somebody to check: the way I get Attributes in ```RateLimitAnnotationListener::getRateLimitsFromAttributes()```. I kind of mixed the ways I found in ```Symfony\Component\HttpKernel\Event\ControllerEvent::setController()```  & ```Symfony\Component\HttpKernel\Event\ControllerEvent::getAttributes()``` (sf6). I should maybe try to use getAttributes() method from ControllerEvent if it exists (sf6) and use my current implementation as a fallback for Sf5.4 ?

I tried to make it the most compatible possible.
Also, I added a condition for uploading coverage in ci but I must verify it still works for non forks.

Let me know what you think.

### CODE REVIEW

**Comment by axi** [comment]
Hum, looks like upload coverage is also broken for main repository...

**Comment by axi** [comment]
Something wrong with that pull request ?

File: EventListener/RateLimitAnnotationListener.php [hunk 1]
```diff
@@ -48,36 +33,39 @@ public function __construct(
...
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
             return;
         }
 
-        // Find the best match
-        $annotations = $event->getRequest()->attributes->get('_x-rate-limit', array());
-        $rateLimit = $this->findBestMethodMatch($event->getRequest(), $annotations);
+        // RateLimits used to be set by sensio/framework-extra-bundle by reading annotations
+        // Tests also use that mechanism, we should probably keep it for retrocompatibility
+        if ($event->getRequest()->attributes->has('_x-rate-limit')) {
```
Comment (goetas) [inline EventListener/RateLimitAnnotationListener.php:50]: nitpick: here you can extract `$event->getRequest()`  in a dedicated `$request` variable

File: EventListener/RateLimitAnnotationListener.php [hunk 2]
```diff
@@ -220,15 +195,30 @@ private function getAliasForRequest($event)
         return 'other';
     }
 
-    private function dispatch($eventName, $event)
+    /**
+     * @return RateLimit[]
+     */
+    public function getRateLimitsFromAttributes(string|array|object $controller): array
```
Comment (goetas) [inline EventListener/RateLimitAnnotationListener.php:201]: this can be private, it is not used anywhere else

**Review by goetas — CHANGES_REQUESTED** [review]
Overall this looks great! just two minor comments on my side

**Review by goetas — APPROVED** [review]

**Comment by goetas** [comment]
@axi  thanks a lot. while solving the conflicts i had to rebase your branch, but now somehow github show my face on the commit despite the fact that the commit is authored by you. :confused:

**Comment by axi** [comment]
@goetas no worries, thanks for mentioning it

**Pull request merged**